docs/specifications/1-introduction.md

```diff
@@ -0,0 +1,172 @@
+# Introduction
+
+This document provides a high level overview of *jsii*, starting with it's
+design tenets. It introduces the concepts and components that compose *jsii*.
+
+
+## Updating the Specification
+
+### Introduction
+
+The *jsii* specification follows the guiding principles of an RFC. It is a
+living document that describes the current understanding of how the various
+[components](#components) of *jsii* are operating, as well as the approaches
+used to ensure consistent behavior across the various supported languages.
+
+The document is mastered with the *jsii* codebase, making it easy to determine
+what specification was in place at the time of a give *jsii* release (by ways of
+referring to the `vX.Y.Z` git tag). A single version of the specification is
+considered **active** at any given time: the version of the specification that
+is represented on the `HEAD` commit of the `master` branch of the [`aws/jsii`]
+repository. The **active** specification must be the base version for any update
+proposal.
+
+[`aws/jsii`]: https://github.com/aws/jsii
+
+The process to update the specification is intended to be as lightweight as
+possible, while ensuring sufficient conversation takes place before implementing
+significant (and breaking) changes. Since the process to update the
+specification is part of the specification itself, it is amenable to be changed
+following the process described in the currently **active** specification.
+
+### Process
+
+While the general process for updating the specification is to create a GitHub
+pull request against the [`aws/jsii`] repository, the exact requirements for
+what should be included in the pull request vary depending on the type of update
+that is proposed:
+
+- [:warning: Changing Behavior](#new-behavior) describes the process to be
+  followed when introducing changes to the behavior of any component of *jsii*:
+  new features, breaking changes to existing features, ...
+- [:mag: Addressing Gaps](#addressing-gaps) is the process used for adding
+  specification around existing but un-specified behavior.
+- [:thumbsup: Trivial Changes](#trivial) explains how to propose changes that
+  improve the specification without changing it's meaning.
+
+#### <a id="new-behavior"/> :warning: Changing Behavior
+
+If the change is **not backwards compatible** (it is a breaking change to an
+existing feature, or it is a new feature that requires all runtime libraries
+implement support immediately), a new RFC should be created in the
+[`awslabs/aws-cdk-rfcs`] repository, following the [RFC Process]. This ensures
+enough time is spent considering alternatives to breaking changes, and to create
+consensus that the change is desirable before time is spent implementing it.
+
+[`awslabs/aws-cdk-rfcs`]: https://github.com/awslabs/aws-cdk-rfcs
+[RFC Process]: https://github.com/aws/aws-cdk-rfcs#what-the-process-is
```

Comment:
Feels like a PR with a diff on the spec can go a long way for describing such changes. Maybe an RFC doc in that case would be accompanied with a PR to the jsii spec instead of having to describe everything twice?

Reply:
If the RFC happens before-hand, then the specification change can be mostly a copy of text from the RFC. But I reckon there is some value in openly stating that a PR on the jsii spec is a valid documentation element for the RFC...